datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1030,6 +1032,9 @@ private void handleDatastreamAddOrDelete() {
             _log.warn("Failed to update datastream: {} after initializing. This datastream will not be scheduled for "
                 + "producing events ", ds.getName());
             shouldRetry = true;
+          } else {
+            // invoke post datastream upsert/delete or state modify actions
```

Comment:
yes, comment is general, I will modify for the current operation. Regarding the datastream being stuck on `INITIALIZING `, I thought about that too. Once the DS is stuck on that, we retry by adding the task in the queue. so when the task gets executed, and if we publish to DH or have any other downstream actions on INITIALIZING, then it will get invoked on retry, which would not be good. You may want to take followup action only once the Datastream gets into READY state, at least that was the intention behind this logic.